presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -0,0 +1,1321 @@
+    {
+        try {
+            return retry()
+                    .stopOn(NoSuchObjectException.class)
+                    .stopOnIllegalExceptions()
+                    .run("getPartitionNamesByParts", stats.getGetPartitionNamesPs().wrap(() ->
+                            getMetastoreClientThenCall(metastoreContext, client -> Optional.of(client.getPartitionNamesFiltered(databaseName, tableName, parts)))));
+        }
+        catch (NoSuchObjectException e) {
+            return Optional.empty();
+        }
+        catch (TException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+        catch (Exception e) {
+            throw propagate(e);
+        }
+    }
+
+    @Override
+    public List<String> getPartitionNamesByFilter(MetastoreContext metastoreContext, String databaseName, String tableName, Map<Column, Domain> partitionPredicates)
+    {
+        List<String> parts = convertPredicateToParts(partitionPredicates);
+        return getPartitionNamesByParts(metastoreContext, databaseName, tableName, parts).orElse(ImmutableList.of());
+    }
+
+    @Override
+    public void addPartitions(MetastoreContext metastoreContext, String databaseName, String tableName, List<PartitionWithStatistics> partitionsWithStatistics)
+    {
+        List<Partition> partitions = partitionsWithStatistics.stream()
+                .map(ThriftMetastoreUtil::toMetastoreApiPartition)
+                .collect(toImmutableList());
+        addPartitionsWithoutStatistics(metastoreContext, databaseName, tableName, partitions);
+        for (PartitionWithStatistics partitionWithStatistics : partitionsWithStatistics) {
+            storePartitionColumnStatistics(metastoreContext, databaseName, tableName, partitionWithStatistics.getPartitionName(), partitionWithStatistics);
+        }
+    }
+
+    private void addPartitionsWithoutStatistics(MetastoreContext metastoreContext, String databaseName, String tableName, List<Partition> partitions)
+    {
+        if (partitions.isEmpty()) {
+            return;
+        }
+        try {
+            retry()
+                    .stopOn(AlreadyExistsException.class, InvalidObjectException.class, MetaException.class, NoSuchObjectException.class, PrestoException.class)
+                    .stopOnIllegalExceptions()
+                    .run("addPartitions", stats.getAddPartitions().wrap(() ->
+                            getMetastoreClientThenCall(metastoreContext, client -> {
+                                int partitionsAdded = client.addPartitions(partitions);
+                                if (partitionsAdded != partitions.size()) {
+                                    throw new PrestoException(HIVE_METASTORE_ERROR,
+                                            format("Hive metastore only added %s of %s partitions", partitionsAdded, partitions.size()));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: partitionsAdded (declared at line 953). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 956. Consider formatting this data with &#39;%d&#39; instead.